tests/tornado/test_connection.py

```diff
@@ -128,3 +129,52 @@ def build_raw_message(self, context, args, is_completed=True):
 
     assert client_peer.total_outbound_pendings == 0
     assert server_peer.total_outbound_pendings == 0
+
+
+@pytest.mark.gen_test
+def test_client_connection_change_callback():
+    server = TChannel('server')
+    server.listen()
+
+    @server.raw.register
+    def hello(request):
+        return 'hi'
+
+    client = TChannel('client')
+    count = [0]
+
+    def test_cb(peer):
+        count[0] += 1
+
+    client._dep_tchannel.peers.get(
+        server.hostport)._on_conn_change_cb = test_cb
```

Comment:
I mean that these tests should be instantiating the Peer class and not accessing it via private variables in the TChannel class. Test the contract of the Peer class. It has the appropriate methods.

You need a series of different tests:

- Tests on the Connection class that assert that the callbacks for the different cases are called as expected. These tests should instantiate the connection class, not the TChannel class.
- Tests on the Peer class that assert that the callback is called as expected. Again, these tests should instantiate the Peer class, call methods on Peer, etc.